src/MultiFormats.Tests/MultiAddressTest.cs

```diff
@@ -0,0 +1,437 @@
+            var good1 = MultiAddress.TryCreate(good.ToArray());
+            Assert.AreEqual(good, good1);
+
+            Assert.IsNull(MultiAddress.TryCreate(new byte[]
+            {
+                0x7f
+            }));
+        }
+
+        [TestMethod]
+        public void JsonSerialization()
+        {
+            var a = new MultiAddress("/ip6/fe80::7573:b0a8:46b0:0bad/tcp/4009");
+            var json = JsonConvert.SerializeObject(a);
+            Assert.AreEqual($"\"{a}\"", json);
+            var b = JsonConvert.DeserializeObject<MultiAddress>(json);
+            Assert.AreEqual(a.ToString(), b.ToString());
+
+            json = JsonConvert.SerializeObject(null);
+            b = JsonConvert.DeserializeObject<MultiAddress>(json);
+            Assert.IsNull(b);
+        }
+
+        [TestMethod]
+        public void WithPeerId()
+        {
+            const string id = "QmQusTXc1Z9C1mzxsqC9ZTFXCgSkpBRGgW4Jk2QYHxKE22";
+            const string id3 = "QmQusTXc1Z9C1mzxsqC9ZTFXCgSkpBRGgW4Jk2QYHxKE33";
+
+            var ma1 = new MultiAddress("/ip4/127.0.0.1/tcp/4001");
+            Assert.AreEqual($"{ma1}/p2p/{id}", ma1.WithPeerId(id));
+
+            ma1 = new MultiAddress($"/ip4/127.0.0.1/tcp/4001/ipfs/{id}");
+            Assert.AreSame(ma1, ma1.WithPeerId(id));
+
+            ma1 = new MultiAddress($"/ip4/127.0.0.1/tcp/4001/p2p/{id}");
+            Assert.AreSame(ma1, ma1.WithPeerId(id));
+
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                ma1 = new MultiAddress($"/ip4/127.0.0.1/tcp/4001/ipfs/{id3}");
+                Assert.AreSame(ma1, ma1.WithPeerId(id));
+            });
+        }
+
+        [TestMethod]
+        public void WithoutPeerId()
+        {
+            var id = "QmQusTXc1Z9C1mzxsqC9ZTFXCgSkpBRGgW4Jk2QYHxKE22";
+
+            var ma1 = new MultiAddress("/ip4/127.0.0.1/tcp/4001");
+            Assert.AreSame(ma1, ma1.WithoutPeerId());
+
+            ma1 = new MultiAddress($"/ip4/127.0.0.1/tcp/4001/ipfs/{id}");
+            Assert.AreEqual("/ip4/127.0.0.1/tcp/4001", ma1.WithoutPeerId());
+
+            ma1 = new MultiAddress($"/ip4/127.0.0.1/tcp/4001/p2p/{id}");
+            Assert.AreEqual("/ip4/127.0.0.1/tcp/4001", ma1.WithoutPeerId());
+        }
+
+        [TestMethod]
+        public void Alias_Equality()
+        {
+            var a = new MultiAddress("/ipfs/QmQusTXc1Z9C1mzxsqC9ZTFXCgSkpBRGgW4Jk2QYHxKE22");
+            var b = new MultiAddress("/p2p/QmQusTXc1Z9C1mzxsqC9ZTFXCgSkpBRGgW4Jk2QYHxKE22");
+
+            Assert.AreEqual(a, b);
+            Assert.IsTrue(a == b);
+            Assert.AreEqual(a.GetHashCode(), b.GetHashCode());
+        }
+
+        [TestMethod]
+        public void GetHashCode_NullValue()
+        {
+            var a = new MultiAddress(
+                "/ip4/139.178.69.3/udp/4001/quic/p2p/QmdGQoGuK3pao6bRDqGSDvux5SFHa4kC2XNFfHFcvcbydY/p2p-circuit/ipfs/QmPJkpfUedzahgVAj6tTUa3DHKVkfTSyvUmnn1USFpiCaF");
+            var _ = a.GetHashCode();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)